src/Bynder/Api/Impl/Upload/FileUploader.php

```diff
@@ -75,300 +49,147 @@ public function __construct(AbstractRequestHandler $requestHandler, AmazonApi $a
-        $eachPromises = new Promise\EachPromise($promises, [
-            'concurrency' => 1,
-            'fulfilled' => function ($pollStatus, $i, $promise) {
-                if ($pollStatus != null) {
-                    if (!empty($pollStatus['itemsDone'])) {
-                        $promise->resolve($pollStatus['itemsDone']);
-                    }
-                    if (!empty($pollStatus['itemsFailed'])) {
-                        $promise->resolve(false);
-                    }
-                }
-                return false;
-            }
-        ]);
-        return $eachPromises->promise();
-    }
-
-    /**
-     * Generates polling promises, sleeping between each request and limiting it to 1 thread to make sure enough time
-     * passes after file upload.
-     *
-     * @param $finalizeResponse
-     *
-     * @return \Generator
-     * @throws Exception
-     */
-    public function pollStatusIterator($finalizeResponse)
-    {
-        $iterations = 0;
-        while ($iterations < self::MAX_POLLING_ITERATIONS) {
-            $delay = $iterations == 0 ? 0 : self::POLLING_IDLE_TIME;
-            yield $this->pollStatusAsync(['items' => $finalizeResponse['importId']], $delay);
-            $iterations++;
-        }
+        $this->requestHandler->sendRequestAsync(
+            'POST',
+            'v7/file_cmds/upload/' . $fileId . '/finalise_api',
+            [
+                'form_params' => $formData
+            ]
+        )->wait();
     }
 
-    /**
-     * Checks if the file has finished uploading.
-     *
-     * @param array $query The import ID of the file to check.
-     *
-     * @param int $delay
-     *
-     * @return Promise\Promise An array of the number successful and failed uploads.
-     * @throws Exception
-     */
-    private function pollStatusAsync($query, $delay = 0)
-    {
-        return $this->requestHandler->sendRequestAsync(
-            'GET',
-            'api/v4/upload/poll/',
-            ['query' => $query, 'delay' => $delay]
-        );
-    }
 
     /**
      * Saves the file in the Bynder Asset Bank. This can be either a new or existing file, depending on whether or not
      * the mediaId parameter is passed.
-     *
-     * @param array $data Array of relevant file upload data, such as uploadId and brandId.
+     * 
+     * @param string $fileId The uuid4 used to identify the file to be uploaded.
+     * @param array $data Array of relevant file upload data, such as brandId, mediaId etc.
      *
      * @return Promise\Promise The information of the uploaded file, including IDs and all final file urls.
      * @throws Exception
      */
-    private function saveMediaAsync($data)
+    private function saveMediaAsync($fileId, $data)
     {
-        $uri = "api/v4/media/save/";
+        if (!isset($data['brandId']) || trim($data['brandId']) === '') {
+            throw new Exception('Invalid or Empty brandId');
+        }
```

Comment:
The brand ID is not needed when there's a media ID. So I'd either make two functions out of this (for the new asset vs. existing asset case), which is probably the cleanest approach.
Or otherwise add a check if, when the brand ID is missing, the media ID is there (which would be a bit more complicated code to grasp).